src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2808,6 +2801,31 @@ void InsertHash(TokenHashValue[] hashTable, String str, TokenType tokenType, int
             } while (i < TOKEN_HASH_SIZE);
             Contract.Assert(true, "The hashtable is full.  This should not happen.");
         }
+
+        private bool CompareStringOptimized(string string1, int offset1, int length1, string string2, int offset2, int length2)
+        {
+            if (length1 != length2)
+            {
+                return false;
+            }
+
+            // Optimize for one character cases which are common due to date and time separators (/ and :)
+            if (length1 == 1)
```

Comment:
Would there be any value in moving this optimization into CompareInfo.CompareString or a similarly centralized location, or do we think the one-character check is too specialized to be used broadly?